drivers/at86rf2xx/at86rf2xx_netdev.c

```diff
@@ -134,13 +150,44 @@ static int _send(netdev_t *netdev, const iolist_t *iolist)
 
     /* send data out directly if pre-loading id disabled */
     if (!(dev->flags & AT86RF2XX_OPT_PRELOADING)) {
-        at86rf2xx_tx_exec(dev);
+        _tigger_send(dev);
     }
     /* return the number of bytes that were actually loaded into the frame
      * buffer/send out */
     return (int)len;
 }
 
+static int _confirm_send(netdev_t *netdev, void *info)
+{
+    (void)info;
+    at86rf2xx_t *dev = (at86rf2xx_t *)netdev;
+    int ret;
+
+    /* ISR sets this */
+    switch (dev->status) {
+        case AT86RF2XX_TRX_STATE__TRAC_SUCCESS:
+        case AT86RF2XX_TRX_STATE__TRAC_SUCCESS_DATA_PENDING:
+            ret = dev->tx_frame_len;
+            break;
+        case AT86RF2XX_TRX_STATE__TRAC_CHANNEL_ACCESS_FAILURE:
+            ret = -EBUSY;
+            break;
+        case AT86RF2XX_TRX_STATE__TRAC_NO_ACK:
+            ret = -ECONNABORTED;
```

Comment:
I see that this is on AVR the case. This is actually quite interesting, since `ECONNABORTED` was only "recently" (POSIX.1-2001), while `ECOMM` has been around for much longer.

In fact, AVR's `errno.h` is pure madness. It allocates two words (16 bit) for an errno, but assigns all but four errno codes to the same value. And two of those have absolutely crazy values, e.g.

```C
/* ((((('E'-64)*26+('N'-64))*26+('O'-64))*26+('S'-64))*26+('Y'-64))*26+'S'-64 */
#define ENOSYS ((int)(66081697 & 0x7fff))

/* (((('E'-64)*26+('I'-64))*26+('N'-64))*26+('T'-64))*26+('R'-64) */
#define EINTR ((int)(2453066 & 0x7fff))
```

People are using `switch()` over errno codes as they believe that `errno` codes (with the documented exception of `EWOULDBLOCK` and `EAGAIN`) have distinct numbers.

I think I'll fix this for RIOT by providing a custom `errno.h` for `cpu/avr8_common`.